src/helpers/getAgreementIdFromIdOrIdcc.js

```diff
@@ -0,0 +1,38 @@
+// @ts-check
+
+const getAgreements = require("../libs/getAgreements");
+
+const INDEXED_AGREEMENTS = getAgreements();
+
+/**
+ * @param {number | string} agreementIdOrIdcc
+ *
+ * @returns {string}
+ */
+function getAgreementIdFromIdOrIdcc(agreementIdOrIdcc) {
+  if (
+    typeof agreementIdOrIdcc === "number" ||
+    (typeof agreementIdOrIdcc === "string" && /^\d{4}$/.test(agreementIdOrIdcc))
+  ) {
+    const idcc = Number(agreementIdOrIdcc);
+    const maybeAgreement = INDEXED_AGREEMENTS.find(({ num }) => num === idcc);
+    if (maybeAgreement === undefined) {
+      throw new Error(`No agreement found with this IDCC (${idcc}).`);
+    }
+
+    return maybeAgreement.id;
+  }
+
+  if (typeof agreementIdOrIdcc === "string" && /^KALICONT\d{12}$/.test(agreementIdOrIdcc)) {
+    const agreementId = agreementIdOrIdcc;
+
+    return agreementId;
```

Comment:
It would probably be safer to use find to and use a different predicate function like (pseudocode)
```js
let findFn = ({num})  => num === idcc;
if (typeof agreementIdOrIdcc === "string" && /^KALICONT\d{12}$/.test(agreementIdOrIdcc)) 
finFn = ({id}) => id === idcc;
const maybeAgreement = INDEXED_AGREEMENTS.find(findFn);
... 
```

Reply:
I also prefer to early throw to unwrap main function code

Reply:
I splitted the code into predicates but I'm not sure to understand the early throwing part. You mean that I should wrap all the functions with a try/catch in order to improve error stacks? I generally do that on final applications but not on libraries as long as they're well test-covered.